File: fathersDay/math/Day1 (1822. Sign of the Product of an Array)/Solution.py
class Solution:
    def arraySign(self, nums: list[int]) -> int:
        product = 1
        for num in nums:
            if num == 0:
                return 0
            elif num < 0:
                product *= -1
        return product
# Example usage
solution = Solution()
print(solution.arraySign([-1, -2, -3, -4, 3, 2, 1])) 
print(solution.arraySign([1, 5, 0, 2, -3]))  
print(solution.arraySign([-1, 1, -1, 1, -1]))  

File: fathersDay/math/Day1 (1822. Sign of the Product of an Array)/Solution.cpp
class Solution {
public:
    int arraySign(std::vector<int>& nums) {
        int productSign = 1;
        for (int num : nums) {
            if (num == 0) {
                return 0;
            } else if (num < 0) {
                productSign *= -1;
            }
        }
        return productSign;
    }
};